hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/ResourceConfigTest.java

```diff
@@ -1024,4 +1033,9 @@ private File createDirectory(String path) {
         assertTrue("Failed to create directory " + dirFile, dirFile.mkdirs());
         return dirFile;
     }
+
+    private static class BaseClass {
+        private static class NestedClass {
```

Comment:
can we add a test with anonymous class too

Reply:
Actually, anonymous class case is not covered yet, I'm working on it right now - will do.

Reply:
Could you also test with Kotlin - since Kotlin lambdas are always created as anonymous classes.

Reply:
Do you mean actually adding kotlin as test dependency or just make sure it works with it? Adding kotlin tests might further slow down the build?

Reply:
Maybe just test it for now - I don't know how complicated it would be to add it as a full dependency and some tests. It's a nice to have.